dotCMS/src/test/java/com/dotcms/util/pagination/SiteViewPaginatorUnitTest.java

```diff
@@ -0,0 +1,211 @@
+        final HostAPI hostAPI = mock(HostAPI.class);
+        long time = System.currentTimeMillis();
+        int i = 0;
+        //System.out.println("Site names: ");
+        for(final String identifier:allSites){
+            final Host host;
+            if(Host.SYSTEM_HOST.equals(identifier)){
+                host = mockSite(identifier, "System Host");
+                when(hostAPI.find(eq(identifier),any(User.class), anyBoolean())).thenReturn(host);
+            } else {
+                final String name = alphabet[i++] + "" +  time;
+                System.out.println(name);
+                host = mockSite(identifier, name);
+            }
+            when(hostAPI.find(eq(identifier),any(User.class), anyBoolean())).thenReturn(host);
+        }
+
+        final List<ContentletSearch> mockedSearch = mockSearchResults(allSites);
+        final ContentletAPI contentletAPI = mock(ContentletAPI.class);
+        when(contentletAPI.searchIndex(anyString(), anyInt(), anyInt(), eq("title"), any(User.class), anyBoolean())).thenReturn(mockedSearch);
+        final Supplier<Set<String>> configuredSitesSupplier = () -> sitesWithIntegrations;
+
+        final SiteViewPaginator paginator = new SiteViewPaginator(configuredSitesSupplier, hostAPI, contentletAPI);
+        final int limit = sitesWithIntegrations.size();
+        final PaginatedArrayList<SiteView> items = paginator
+                .getItems(user, null, limit, 0, null, null, Collections.emptyMap());
+
+        Assert.assertNotNull(items);
+        Assert.assertFalse(items.isEmpty());
+        Assert.assertEquals(items.get(0).getId(), Host.SYSTEM_HOST);
+        Assert.assertEquals(items.size(), limit);
+        //First item is'nt necessarily configured. So we start counting from 1.
+        for(int j=1; j < limit; j++){
+            Assert.assertTrue(items.get(j).isConfigured());
+        }
+    }
+
+    @Test
+    public void Test_Get_Items_Sorted_Pages()
+            throws DotSecurityException, DotDataException {
+        final User user = mockAdminUser();
+        final int maxConfigured = 6; //Only the first page is expected to bring back configured items.
+        final List<String> allSites = mockAllSitesIdentifiers(alphabet.length - 1);
+        final HostAPI hostAPI = mock(HostAPI.class);
+        final List<String> allSitesSortedIdentifiers = new LinkedList<>();
+        final long time = System.currentTimeMillis();
+        //final Map<String,String> debugInfo = new TreeMap<>();
+        int i = 0;
+        //System.out.println("Site names: ");
+        for(final String identifier:allSites){
+            final Host host;
+            final String name;
+            if(Host.SYSTEM_HOST.equals(identifier)){
+               name = "System Host";
+               host = mockSite(identifier, name);
+               when(hostAPI.find(eq(identifier),any(User.class), anyBoolean())).thenReturn(host);
+            } else {
+               name = alphabet[i++] + "" +  time;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not add empty strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4960082)